openpype/hosts/maya/plugins/inventory/connect_yeti_rig.py

```diff
@@ -0,0 +1,156 @@
+        ]
+        message = (
+            "{} animation containers selected:\n\n{}\n\nOnly select 1 of type "
+            "\"animation\" or \"pointcache\".".format(
+                len(source_containers), source_container_namespaces
+            )
+        )
+        if len(source_containers) != 1:
+            self.display_warning(message)
+            return
+
+        source_container = source_containers[0]
+        source_ids = self.nodes_by_id(source_container)
+
+        # Target containers.
+        target_ids = {}
+        inputs = []
+        for family, containers in containers_by_family.items():
+            if family != "yetiRig":
+                continue
+
+            for container in containers:
+                target_ids.update(self.nodes_by_id(container))
+
+                maya_file = get_representation_path(
+                    get_representation_context(
+                        container["representation"]
+                    )["representation"]
+                )
+                _, ext = os.path.splitext(maya_file)
+                settings_file = maya_file.replace(ext, ".rigsettings")
+                if not os.path.exists(settings_file):
+                    continue
+
+                with open(settings_file) as f:
+                    inputs.extend(json.load(f)["inputs"])
+
+        # Compare loaded connections to scene.
+        for input in inputs:
+            source_node = source_ids.get(input["sourceID"])
+            target_node = target_ids.get(input["destinationID"])
+
+            if not source_node or not target_node:
+                self.log.debug(
+                    "Could not find nodes for input:\n" +
+                    json.dumps(input, indent=4, sort_keys=True)
+                )
+                continue
+            source_attr, target_attr = input["connections"]
+
+            if not cmds.attributeQuery(
+                source_attr, node=source_node, exists=True
+            ):
+                self.log.debug(
+                    "Could not find attribute {} on node {} for "
+                    "input:\n{}".format(
+                        source_attr,
+                        source_node,
+                        json.dumps(input, indent=4, sort_keys=True)
+                    )
+                )
+                continue
+
+            if not cmds.attributeQuery(
+                target_attr, node=target_node, exists=True
+            ):
+                self.log.debug(
+                    "Could not find attribute {} on node {} for "
+                    "input:\n{}".format(
+                        target_attr,
+                        target_node,
+                        json.dumps(input, indent=4, sort_keys=True)
+                    )
+                )
+                continue
+
+            cmds.connectAttr(
+                "{}.{}".format(source_node, source_attr),
+                "{}.{}".format(target_node, target_attr)
+            )
```

Comment:
I'm thinking now - what if you've updated your yetiRig which should now make new extra connections. You can't re-run this action because it'll error on some attributes already being connected. Should we first check if they are connected?

Plus we'd want to force override the connection if the target was already connected to another source.
```suggestion

            source_plug = "{}.{}".format(source_node, 
                                         source_attr)
            target_plug = "{}.{}".format(target_node, 
                                         target_attr)
            if cmds.isConnected(source_plug, 
                                target_plug,
                                ignoreUnitConversion=True):
                self.log.debug(
                    "Connection already exists: {} -> {}".format(
                        source_plug, target_plug
                    )
                )
                continue

            cmds.connectAttr(source_plug, target_plug, force=True)
            self.log.debug(
                "Connected attributes: {} -> {}".format(
                    source_plug, target_plug
                )
            )
```